### ISSUE TITLE
Geo Preview: fail gracefully on ludicrous geometry?

### ISSUE THREAD

**Issue by cjhendrix** [OCHA-DAP/hdx-ckan#3280]
Summarizing the issue here.  More details are below.

It seems that some systems use very large negative numbers for null geometries.  This are handled fine by PostGIS, but they break Leaflet.

I propose that we convert out-of-range coordinates to 0,0.  This will allow them to preview and look strange.  It will highlight to the contributor that there is a problem with their data that should be fixed. 

One potential problem may be with non-point geometries.  0,0 for all verticies in a polygon or line may also cause a problem. We will need to experiment with that.

---

Something funny is happening with this one.  The tranformation completes successfully.  But the page won't load and chrome eventually kills it.  Same on prod and on stag.  Also the same after downloading the file and resaving to a new shapefile in qgis. 

https://test-data.hdx.rwlabs.org/api/3/action/package_show?id=syria-cultural-sites

The lower-left coordinate of the bounding box looks weird. 

"bounding_box": "BOX(-1.79769313486232e+308 -1.79769313486232e+308,41.9559722222222 37.0577083333333)"

 And I noticed in qgis that there is one record with NULL for the LAT and LONG columns (these are not the shapefile geometry, just a string field).  But when I do expose the coordinates, this point (Facility_N = "Damascus, Army Museum") is at  

POINT(-179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368 -179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368)

I don't know where that coordinate is, but it is literally probably not in this universe.  

So, the real issue here is, how can we fail more gracefully on absurd geometries?

**Comment by cjhendrix** [comment]
This one may be another example:  https://data.hdx.rwlabs.org/dataset/honduras-other-0-0  

(note that the resource has been "fixed" so the problem of crashing the browser is no longer an issue on this particular dataset, but the overall problem remains, read on)

This is the same problem. There are 12 points in the dataset that have coordinates = 12 records (points) in that dataset that have coordinates = -1.7976931348623157e+308, -1.7976931348623157e+308.

Exactly the same as the syria example in the main issue description above. This is probably some default that ESRI generates in certain conditions for records that have no geometry. I wonder if we could get our converter to change them to 0,0 That would make the error quite obvious in our geopreview (and stop it from crashing the browser!).

**Comment by cjhendrix** [comment]
Yes, this is the same problem. There are 12 points in the dataset that have coordinates = 12 records (points) in that dataset that have coordinates = -1.7976931348623157e+308, -1.7976931348623157e+308.

Exactly the same as the syria example in the main issue description above.  This is probably some default that ESRI generates in certain conditions for records that have no geometry. I wonder if we could get our converter to change them to 0,0  That would make the error quite obvious in our geopreview  (and stop it from crashing the browser!).

**Comment by teodorescuserban** [comment]
I wonder how 1e+308 looks like in js, python or postgres :)

Mie, 21 oct. 2015, 18:49, cjhendrix notifications@github.com a scris:

**Comment by teodorescuserban** [comment]
I mean, jokes aside, we can just replace the numbers greater than longint
with 0 in js - addressing the browser issue right in the browser.

Because other chain parts doesnt seem to have a problem with it.

Mie, 21 oct. 2015, 20:33, Serban Teodorescu teodorescu.serban@gmail.com a
scris:

**Comment by teodorescuserban** [comment]
Some solution around replacing the string e+xxx with 0 in js before we even
try to reder anything.
I say string because number compare might lead to the same issue.

Mie, 21 oct. 2015, 20:35, Serban Teodorescu teodorescu.serban@gmail.com a
scris:

**Comment by cjhendrix** [comment]
Similar problem when uploading this file: https://www.dropbox.com/s/rmegntsfq7l275i/malawi_health_fac_hisp.zip?dl=0

The extent of the shapefile is correct, however the bbox visible in the ckan api has the magic 1+e308 and in HDX, the page never completes loading.

In this case, the input layer has a partially defined coordinate system (WGS84).    I re-exported with epsg:4326 and it works correctly on HDX now. https://data.hdx.rwlabs.org/dataset/malawi-health
